pkg/receive/receive_test.go

```diff
@@ -5,10 +5,220 @@ package receive
+				},
+				{
+					Endpoints:      []string{"node2"},
+					Tenants:        []string{"tenant4", "tenant5", "tenant6"},
+					ExternalLabels: []string{"name4=value4", "name5=value5", "name6=value6"},
+				},
+			},
+			expectedLabelSets: []labels.Labels{
+				labels.FromStrings("replica", "test", "tenant_id", "tenant1",
+					"name1", "value1", "name2", "value2", "name3", "value3"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant2",
+					"name1", "value1", "name2", "value2", "name3", "value3"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant3",
+					"name1", "value1", "name2", "value2", "name3", "value3"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant4",
+					"name4", "value4", "name5", "value5", "name6", "value6"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant5",
+					"name4", "value4", "name5", "value5", "name6", "value6"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant6",
+					"name4", "value4", "name5", "value5", "name6", "value6"),
+			},
+		},
+		{
+			name: "Multiple hashrings - One repeated tenant",
+			cfg: []HashringConfig{
+				{
+					Endpoints:      []string{"node1"},
+					Tenants:        []string{"tenant1", "tenant2", "tenant3"},
+					ExternalLabels: []string{"name1=value1", "name2=value2", "name3=value3"},
+				},
+				{
+					Endpoints:      []string{"node2"},
+					Tenants:        []string{"tenant4", "tenant5", "tenant1"},
+					ExternalLabels: []string{"name4=value4", "name5=value5", "name6=value6"},
+				},
+			},
+			expectedLabelSets: []labels.Labels{
+				labels.FromStrings("replica", "test", "tenant_id", "tenant1",
+					"name1", "value1", "name2", "value2", "name3", "value3"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant2",
+					"name1", "value1", "name2", "value2", "name3", "value3"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant3",
+					"name1", "value1", "name2", "value2", "name3", "value3"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant4",
+					"name4", "value4", "name5", "value5", "name6", "value6"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant5",
+					"name4", "value4", "name5", "value5", "name6", "value6"),
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			var bucket objstore.Bucket
+
+			m := NewMultiTSDB(t.TempDir(), log.NewNopLogger(), prometheus.NewRegistry(),
+				&tsdb.Options{
+					MinBlockDuration:  (2 * time.Hour).Milliseconds(),
+					MaxBlockDuration:  (2 * time.Hour).Milliseconds(),
+					RetentionDuration: (6 * time.Hour).Milliseconds(),
+				},
+				labels.FromStrings("replica", "test"),
+				"tenant_id",
+				bucket,
+				false,
+				metadata.NoneFunc,
+			)
+
+			err := m.SetHashringConfig(tc.cfg)
+			require.NoError(t, err)
+
+			for _, c := range tc.cfg {
+				for _, tenantId := range c.Tenants {
+					m.TenantAppendable(tenantId)
+				}
+			}
```

Comment:
I think we might need something like this, to get tenant data to populate. 
```suggestion
			for _, c := range tc.cfg {
				for _, tenantId := range c.Tenants {
				        testutil.Ok(t, appendSample(m, tenantId, time.Now()))
				}
			}
```

Then you can check that the stores exist as well as the labels, via something like,
```go
clients := m.TSDBLocalClients()

testutil.Equals(t, numberOfTenants, len(clients))

for i := range clients {
	testutil.Equals(t, tc.expectedLabels, clients[i].LabelSets())
}
```